unitTests/ui/MissingTermsTest.ts

```diff
@@ -0,0 +1,192 @@
+        const caption = 'The missing term';
+        test = mockComponent({ caption });
+        const missingTermsElement = $$(test.cmp.element).find('.coveo-field-caption');
+        expect(missingTermsElement.innerText).toBe(caption);
+      });
+    });
+    describe('when the langage is English', () => {
+      beforeEach(() => {
+        test = mockComponent();
+      });
+      describe('when fetching the missing terms from a query', () => {
+        describe('return', () => {
+          let expectedResult: string[];
+          beforeEach(() => {
+            expectedResult = [...fakeResult.absentTerms];
+          });
+          it('all the missing term present in the query', () => {
+            fakeResult.absentTerms.push(termIsSubWord);
+            expectedResult.push(termIsSubWord);
+            expect(test.cmp.missingTerms.toString()).toBe(expectedResult.toString());
+          });
+
+          it('only the missing term present in the query', () => {
+            fakeResult.absentTerms.push(termNotInQuery);
+            expect(test.cmp.missingTerms.toString()).toBe(expectedResult.toString());
+          });
+        });
+      });
+
+      describe('when re-injecting a term as an exact match', () => {
+        it('and the term is present in the query, the term is re-injected as an exact match', () => {
+          test.cmp.includeTermInQuery(termPresent);
+          expect(test.cmp.queryStateModel.set).toHaveBeenCalledWith(
+            'q',
+            'This is and "will" be a really really really-really long query!!!'
+          );
+        });
+
+        it('and the term is the first word in the query, the term is re-injected as an exact match', () => {
+          const firstWord = 'This';
+          fakeResult.absentTerms.push(firstWord);
+          test.cmp.includeTermInQuery(firstWord);
+          expect(test.cmp.queryStateModel.set).toHaveBeenCalledWith(
+            'q',
+            '"This" is and will be a really really really-really long query!!!'
+          );
+        });
+
+        it('and the term is the last word in the query, the term is re-injected as an exact match', () => {
+          const lastWord = 'query';
+          fakeResult.absentTerms.push(lastWord);
+          test.cmp.includeTermInQuery(lastWord);
+          expect(test.cmp.queryStateModel.set).toHaveBeenCalledWith(
+            'q',
+            'This is and will be a really really really-really long "query"!!!'!!!
+          );
+        });
+
+        it('and the term is present multiple times in the query, the term are all re-injected as an exact match', () => {
+          const wordPresentMultipleTimes = 'really';
+          fakeResult.absentTerms.push(wordPresentMultipleTimes);
+          test.cmp.includeTermInQuery(wordPresentMultipleTimes);
+          expect(test.cmp.queryStateModel.set).toHaveBeenCalledWith(
+            'q',
+            'This is and will be a "really" "really" really-really long query!!!'
```

Comment:
I think these tests are hard to follow because they reuse the same setup.

When I write unit tests, I try as much as possible to use the most minimal setup required to display the issue/functionality. In this case, it would be nice to see the English Spy created in the test along with it's return value. e.g.

```
 const getEnglishSpy: any = jasmine
      .createSpy('getSpy')
      .and.returnValue('really really');
```
Also, instead of referencing the `termIsSubWord`, `termNotInQuery` etc., you can see those values inside the test directly. This makes the tests independent of one another, easier to follow, and later on easier to expand since we are not modifying the global setup any more.

Reply:
ok so I can do something like 
```
it('and the term is present multiple times in the query, the term are all re-injected as an exact match', () => {
    const wordPresentMultipleTimes = 'really';
    getEnglishSpy.and.returnValue('This is really really my query');
    fakeResult.absentTerms = [wordPresentMultipleTimes];
    test.cmp.includeTermInQuery(wordPresentMultipleTimes);
    expect(test.cmp.queryStateModel.set).toHaveBeenCalledWith(
      'q',
      'This is "really" "really" my query'
    );
 });
```
Is it ok? 
My English spy will have the same setup a lot of the time but yeah it is a lot easier to read

Reply:
I find this much easier to read too. Personally I would go even shorter with the spy return value, but this is fine too.